### PR TITLE
DOC: Mention that expand_dims and squeeze are inverses

### PR DESCRIPTION
**Pull request by eric-wieser**
[ci skip]'d

If #8735 might not be such a good idea, this gets most of the way there.

### CODE REVIEW

File: numpy/lib/shape_base.py [hunk 1]
```diff
@@ -255,6 +255,7 @@ def expand_dims(a, axis):
 
     See Also
     --------
+    squeeze : The inverse operation, removing singleton dimensions
```
Comment (charris) [inline numpy/lib/shape_base.py]: Maybe, "removes all singleton dimensions".

Comment (eric-wieser) [inline numpy/lib/shape_base.py]: What I mean here is that `np.squeeze(arr, axis=a)` is the inverse of `np.expand_dims(arr, axis=a)`. `squeeze`

Comment (charris) [inline numpy/lib/shape_base.py]: Might also mention `reshape` in both contexts as it allows adding and removing axis.

Comment (seberg) [inline numpy/lib/shape_base.py]: I personally like indexing a lot for this both of this too, unless it gets very clunky with counting `:`.

Comment (eric-wieser) [inline numpy/lib/shape_base.py]: @charris: Added comments about reshape

**Comment by charris** [comment]
Thanks Eric.

**Pull request merged**